### ISSUE TITLE
"build coreclr" rebuilds everything every time

### ISSUE THREAD

**Issue by dsyme** [Microsoft/visualfsharp#3295]
Previously 

    build coreclr

was clean on a second build.  Now it rebuilds everything, doing the build from source, making incremental build & test very slow for contributors

**Comment by dsyme** [comment]
Also ``build coreclr test debug`` doesn't seem to test the debug build cleanly, as IO see ``testbin\release...`` being used in the build:
```
C:\GitHub\dsyme\visualfsharp\Tools\dotnetcli\dotnet.exe C:\GitHub\dsyme\visualfsharp\tests\testbin\release\coreclr\fsc\fsc.exe --noframework ...
```

**Comment by dsyme** [comment]
Closing old issue, fixed by #4273 if not before
